training-tracker-service/src/main/java/org/tolking/messaging/TrackerEventConsumer.java

```diff
@@ -0,0 +1,32 @@
+package org.tolking.messaging;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.slf4j.MDC;
+import org.springframework.jms.annotation.JmsListener;
+import org.springframework.messaging.handler.annotation.Header;
+import org.springframework.stereotype.Component;
+import org.tolking.service.TrainingEventService;
+import org.tolking.external_dto.TrainingEventDTO;
+
+import static org.tolking.filter.LoggingFilter.CORRELATION_ID;
+
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class TrackerEventConsumer {
+    private final TrainingEventService trainingEventService;
+    private final static String QUEUE_NAME = "training-event-queue";
+
+    @JmsListener(destination = QUEUE_NAME)
+    public void receiveMessage(TrainingEventDTO trainingEventDTO, @Header(CORRELATION_ID) String correlationId) {
+        addTraceId(correlationId);
+
+        log.debug("Received message from queue {} : {}", QUEUE_NAME, trainingEventDTO);
+        trainingEventService.create(trainingEventDTO);
```

Comment:
what will happen if message was invalid? wrap the code with try-catch. Usually, if exception occur message will come back to queue if retriable or to dead letter queue.